File: 349-Intersection-of-Two-Arrays/go/main.go
package main

import "fmt"

func intersection(nums1 []int, nums2 []int) []int {
    m := make(map[int]bool)
    var res []int

    for _, num := range nums1 {
        m[num] = true
    }

    for _, num := range nums2 {
        if m[num] {
            res = append(res, num)
            delete(m, num)
        }
    }

    return res
}

func main() {
    nums1 := []int{1, 2, 2, 1}
    nums2 := []int{2, 2}
    fmt.Println(intersection(nums1, nums2))
}

File: 349-Intersection-of-Two-Arrays/rs/src/main.rs
fn main() {
    println!("Hello, world!");
}



use std::collections::HashSet;

pub fn intersection(nums1: Vec<i32>, nums2: Vec<i32>) -> Vec<i32> {
    let set1: HashSet<_> = nums1.into_iter().collect();
    let set2: HashSet<_> = nums2.into_iter().collect();
    set1.intersection(&set2).cloned().collect()
}
